content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,738 @@
+
+```json
+{
+  "nodeId": 0, // go-waku node that emitted the signal
+  "type": "message", // type of signal being emitted. Currently only "message" is available
+  "event": ... // format depends on the type of signal. In the case of "message", a waku message can be expected.
+}
+```
+
+| `type`    | `event` Type       |
+|:----------|--------------------|
+| `message` | `JsonMessageEvent` | 
+
+### JsonMessageEvent
+
+Type of `event` field for a `message` event:
+
+```ts
+{
+    subscriptionId: number;
+    pubsubTopic: string;
+    messageId: string;
+    wakuMessage: JsonMessage;
+  }
+```
+
+- `subscriptionId`: The id of the subscription from which the message was received, returned by [`waku_relay_subscribe`](#extern-char-waku_relay_subscribeint-nodeid-char-pubsubtopic).
+- `pubsubTopic`: The pubsub topic on which the message was received.
+- `messageId`: The message id.
+- `wakuMessage`: The message in [`JsonMessage`](#jsonmessage) format.
+
+### `extern void waku_set_event_callback(void* cb)`
+
+Register callback to act as event handler and receive application signals,
+which are used to react to asynchronous events in Waku.
+
+**Parameters**
+
+1. `void* cb`: callback that will be executed when an async event is emitted.
+  The function signature for the callback should be `void myCallback(char* signalJSON)`
+
+## Node management
+
+### `extern char* waku_new(char* configJson)`
+
+Instantiates a Waku node.
+
+**Parameters**
+
+1. `char* configJson`: JSON string containing the options used to initialize a go-waku node.
+   It can be `NULL` to use defaults.
+   All keys are optional.
+   If a key is `undefined`, or `null`, a default value will be set. Example:
+   ```json
+   {
+     "host": "0.0.0.0",
+     "port": 60000,
+     "advertiseAddr": "1.2.3.4",
+     "nodeKey": "0x123...567",
+     "keepAliveInterval": 20,
+     "relay": true,
+     "minPeersToPublish": 0
+   }
+   ```
+    - `host` - `string` (optional): Listening IP address.
+      Default `0.0.0.0`.
+    - `port` - `number` (optional): Libp2p TCP listening port.
+      Default `60000`.
+      Use `0` for random.
+    - `advertiseAddr` - `string` (optional): External address to advertise to other nodes.
+    - `nodeKey` - `string` (optional): secp256k1 private key in Hex format (`0x123...abc`).
+      Default random.
+    - `keepAliveInterval` - `number` (optional): Interval in seconds for pinging peers to keep the connection alive.
+      Default `20`.
+    - `relay` - `boolean` (optional): Enable relay protocol.
+      Default `true`.
+    - `minPeersToPublish` - `number` (optional).
+      The minimum number of peers required on a topic to allow broadcasting a message.
+      Default `0`.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains an `integer` which represents the `nodeId`.
+This node id MUST be used in following API calls to interact with the instantiated node.
```

Comment:
It's possible to create multiple node instantantiations via the API. But if we agree to only allow a single instance, we can remove this node Id parameter, and assume only a single node can be used

Reply:
Afaict if we instantiate a single node through the API and can get rid of `subscriptionId` for `pubsubTopic`s elsewhere, we may be able to define a stateless API.

Reply:
I like this idea. I don't see a strong need for multiple nodes. We could still re-introduce it there is a need for it but I am not sure it would make sense.

Reply:
Let's remove it for now.

Reply:
Sounds good. I removed it also from go-waku C binding, C# and Kotlin examples